test/muxer_test.exs

```diff
@@ -0,0 +1,241 @@
+    end
+  end
+
+  defmodule TestPipelineMany do
+    use Membrane.Pipeline
+
+    @impl true
+    def handle_init(options) do
+      opus_source = %Testing.Source{
+        output: Testing.Source.output_from_buffers(options.buffers),
+        caps: %Opus{channels: 2, self_delimiting?: false}
+      }
+
+      children = [
+        vpx_source: %Membrane.File.Source{
+          location: options.input_file
+        },
+        deserializer: Membrane.Element.IVF.Deserializer,
+        opus_source: opus_source,
+        muxer: Membrane.WebM.Muxer,
+        sink: %Membrane.File.Sink{
+          location: options.output_file
+        },
+        parser: Membrane.WebM.Parser,
+        printer: Printer
+      ]
+
+      links = [
+        link(:vpx_source)
+        |> to(:deserializer)
+        |> via_in(Pad.ref(:input, 11_020_961_587_148_742_657))
+        |> to(:muxer),
+        link(:opus_source)
+        |> via_in(Pad.ref(:input, 16_890_875_709_512_990_721))
+        |> to(:muxer),
+        link(:muxer)
+        # |> to(:parser)
+        # |> to(:printer)
+        |> to(:sink)
+      ]
+
+      {{:ok, spec: %ParentSpec{children: children, links: links}}, %{}}
+    end
+  end
+
+  defp test_from_buffers(tmp_dir) do
+    buffers =
+      (@fixtures_dir <> "buffers_dump.opus")
+      |> File.read!()
+      |> :erlang.binary_to_term()
+      |> Enum.reverse()
+
+    {:ok, pipeline} =
+      %Testing.Pipeline.Options{
+        module: TestPipelineOpus,
+        custom_args: %{
+          output_file: tmp_dir <> "output.webm",
+          buffers: buffers
+        }
+      }
+      |> Testing.Pipeline.start_link()
+
+    play_and_validate(pipeline, "muxed_opus.webm", tmp_dir)
+  end
+
+  defp test_stream(input_file, reference, tmp_dir) do
+    {:ok, pipeline} =
+      %Testing.Pipeline.Options{
+        module: TestPipelineVpx,
+        custom_args: %{
+          input_file: @fixtures_dir <> input_file,
+          output_file: tmp_dir <> "output.webm"
+        }
+      }
+      |> Testing.Pipeline.start_link()
+
+    play_and_validate(pipeline, reference, tmp_dir)
+  end
+
+  defp test_many(tmp_dir) do
+    buffers =
+      (@fixtures_dir <> "buffers_dump.opus")
+      |> File.read!()
+      |> :erlang.binary_to_term()
+      |> Enum.reverse()
+
+    {:ok, pipeline} =
+      %Testing.Pipeline.Options{
+        module: TestPipelineMany,
+        custom_args: %{
+          input_file: @fixtures_dir <> "1_vp8.ivf",
+          output_file: tmp_dir <> "output.webm",
+          buffers: buffers
+        }
+      }
+      |> Testing.Pipeline.start_link()
+
+    play_and_validate(pipeline, "combined.webm", tmp_dir)
+  end
+
+  defp play_and_validate(pipeline, output_file_name, tmp_dir) do
+    Testing.Pipeline.play(pipeline)
+    assert_pipeline_playback_changed(pipeline, _, :playing)
+    assert_start_of_stream(pipeline, :sink, :input, 100_000)
+    assert_end_of_stream(pipeline, :sink, :input, 100_000)
```

Comment:
This is way too long. The test should really finish after 2 seconds, 10 is an absolute max. Your files are probably too big, or we have serious performance issues